site/src/pages/guides/adding-a-second-entrypoint.mdx

```diff
@@ -0,0 +1,110 @@
+# Adding a second entrypoint
+
+When adding a second entrypoint, you are creating a second place where a package can be accessed from. Note that much of the time you do not need a second entrypoint. We have a guide on [when to use multiple entrypoints](/guides/when-should-i-use-multiple-entrypoints).
+
+So, let's assume we have a package called `emoji-uid` that gets us three random emojis, but we want to make a second function that only uses the best emojis. What we want to be able to do is write is:
+
+```js
+import getEmojiId from "emoji-uid";
+import getAwesomeId from "emoji-uid/awesome";
+```
+
+With `preconstruct`'s base setup, this isn't possible, so we will need to add a new `awesome` entrypoint.
+
+First, our project structure looks like:
+
+```
+- src
+    /index.js
+- package.json
+```
+
+and our package.json looks like:
+
+```json
+{
+  "name": "emoji-uid",
+  "version": "1.0.2",
+  "description": "Get 🔥 UIDs to help make things 💯",
+  "main": "dist/emoji-uid/.cjs.js",
+  "preconstruct": {
+    "source": "src/index.js"
+  }
+}
+```
+
+Let's assume we've written our new function and written it to `src/awesomenewFunction.js`
+
+The first bit of config we need to do is to add an `entrypoints` field to our preconstruct config. We should modify this in our package.json to read:
+
+```json
+  "preconstruct": {
+    "source": "src/index.js"
+    "entrypoints": [".", "awesome"]
+  },
+```
+
+This tells preconstruct to look for these two entrypoints. We have written out two, as the first is preconstruct's default entrypoint which we still want to include.
+
+Next, we will want to do to add the location the entrypoint resolves to. We are going to add a new directory with its own `package.json` that represents this. We can run:
+
+```bash
+mkdir awesome
+touch awesome/package.json
+```
```

Comment:
Not about this PR but btw, I think I have a way to make it so you don't need to create the directory but globs still work.

Reply:
😆 how much mad science is it? How scared am I?

Super great though.

Reply:
It's not _that_ much mad science. The assumption is that a glob in an array of globs that isn't really a glob(it's just a path) should either match something or shouldn't be in the array of globs. If we use this assumption, we can go through the array of globs and check if it's a glob or not, if it's not, make sure that that the path/glob matches against the array of globs(just in case it's negated) and then check if the directory/package.json and if it doesn't prompt to create the entrypoint(and if they decline, remove the glob thing from the array of globs)